panther_docking/include/panther_docking/panther_charging_dock.hpp

```diff
@@ -0,0 +1,243 @@
+   */
+  std::string getName();
+
+protected:
+  /**
+   * @brief Method calls enable/disable service of the charger
+   *
+   * @param state The state to set the charger to
+   */
+  void setChargerState(bool state);
+
+  /**
+   * @brief Transform a pose to a target frame.
+   *
+   * @param pose The pose to transform.
+   * @param target_frame The target frame to transform the pose to.
+   *
+   * @return The transformed pose.
+   */
+  geometry_msgs::msg::PoseStamped transformPose(
+    const geometry_msgs::msg::PoseStamped & pose, const std::string & target_frame);
+
+  /**
+   * @brief Offset the staging pose.
+   *
+   * This method offsets the staging dock pose by values described in a configuration.
+   *
+   * @param dock_pose The dock pose to offset.
+   *
+   * @return The offset staging pose.
+   */
+  geometry_msgs::msg::PoseStamped offsetStagingPoseToDockPose(
+    const geometry_msgs::msg::PoseStamped & dock_pose);
+
+  /**
+   * @brief Offset the detected dock pose.
+   *
+   * This method offsets the detected dock pose by values described in a configuration.
+   *
+   * @param detected_dock_pose The detected dock pose to offset.
+   *
+   * @return The offset detected dock pose.
+   */
+  geometry_msgs::msg::PoseStamped offsetDetectedDockPose(
+    const geometry_msgs::msg::PoseStamped & detected_dock_pose);
+
+  /**
+   * @brief Get the dock pose in a detection dock frame.
+   *
+   * This method retrieves the dock pose in a detection dock frame.
+   *
+   * @param frame The detection frame to get the dock pose in.
+   * @return The dock pose in the detection frame.
+   */
+  geometry_msgs::msg::PoseStamped getDockPose(const std::string & frame);
+
+  /**
+   * @brief Method to update the dock pose and publish it.
+   *
+   * This method makes all necessary transformations to update the dock pose and publishes it.
+   *
+   */
+  void updateDockPoseAndPublish();
+
+  /**
+   * @brief Update the staging pose and publish it.
+   *
+   * This method makes all necessary transformations to update the staging pose and publishes it.
+   *
+   * @param frame The frame to publish the staging pose in.
+   */
+  void updateStagingPoseAndPublish(const std::string & frame);
+
+  std::string name_;
+  std::string base_frame_name_;
+  std::string dock_frame_;
+  double panther_version_;
+
+  rclcpp::Logger logger_{rclcpp::get_logger("PantherChargingDock")};
+  rclcpp::Clock steady_clock_{RCL_STEADY_TIME};
+  rclcpp_lifecycle::LifecycleNode::SharedPtr node_;
+  std::shared_ptr<tf2_ros::Buffer> tf2_buffer_;
+  rclcpp::Subscription<panther_msgs::msg::ChargingStatus>::SharedPtr charging_status_sub_;
+  rclcpp::Subscription<panther_msgs::msg::IOState>::SharedPtr io_state_sub_;
+  rclcpp::Publisher<geometry_msgs::msg::PoseStamped>::SharedPtr staging_pose_pub_;
+  rclcpp::Publisher<geometry_msgs::msg::PoseStamped>::SharedPtr dock_pose_pub_;
+  rclcpp::Client<std_srvs::srv::SetBool>::SharedPtr charger_enable_client_;
```

Comment:
Vertical segregation is very welcomed here to improve readability.

```suggestion
  rclcpp::Logger logger_{rclcpp::get_logger("PantherChargingDock")};
  rclcpp::Clock steady_clock_{RCL_STEADY_TIME};
  
  rclcpp_lifecycle::LifecycleNode::SharedPtr node_;
  std::shared_ptr<tf2_ros::Buffer> tf2_buffer_;
  
  rclcpp::Subscription<panther_msgs::msg::ChargingStatus>::SharedPtr charging_status_sub_;
  rclcpp::Subscription<panther_msgs::msg::IOState>::SharedPtr io_state_sub_;
  
  rclcpp::Publisher<geometry_msgs::msg::PoseStamped>::SharedPtr staging_pose_pub_;
  rclcpp::Publisher<geometry_msgs::msg::PoseStamped>::SharedPtr dock_pose_pub_;
  
  rclcpp::Client<std_srvs::srv::SetBool>::SharedPtr charger_enable_client_;
```